mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js

```diff
@@ -0,0 +1,199 @@
+    init: function(record) {
+
+        // create layer : container of geometry to display adress selected
+        this.layer = new OpenLayers.Layer.Vector("__georchestra_bangeocoder", {
+            displayInLayerSwitcher: false,
+            styleMap: new OpenLayers.StyleMap({
+                "default": {
+                    graphicWidth: 20,
+                    graphicHeight: 32,
+                    graphicYOffset: -28, // shift graphic up 28 pixels
+                    //Change the location of picture if not find
+                    externalGraphic: GEOR.config.PATHNAME + 'app/css/images/pwrs/geoPin.png',
+                }
+            })
+        });
+
+        // create main window containing free text combo
+        this.addressField = this._createCbSearch();
+        this.win = new Ext.Window({
+            title: OpenLayers.i18n('banGeocoder.window_title'),
+            constrainHeader: true,
+            width: 312,
+            closable: true,
+            closeAction: "hide",
+            resizable: false,
+            border: false,
+            cls: "bangeocoder",
+            items: [this.addressField],
+            listeners: {
+                "hide": function() {
+                    this.map.removeLayer(this.layer);
+                    this.item && this.item.setChecked(false);
+                    this.components && this.components.toggle(false);
+                },
+                "show": function() {
+                    this.map.addLayer(this.layer);
+                },
+                scope: this
+            }
+        });
+
+        if (this.target) {
+            // create a button to be inserted in toolbar:
+            this.components = this.target.insertButton(this.position, {
+                xtype: 'button',
+                tooltip: this.getTooltip(record),
+                iconCls: 'addon-bangeocoder',
+                handler: this.showWindow,
+                scope: this
+            });
+            this.target.doLayout();
+        } else {
+            // create a menu item for the "tools" menu:
+            this.item = new Ext.menu.CheckItem({
+                text: this.getText(record),
+                qtip: this.getQtip(record),
+                iconCls: "addon-bangeocoder",
+                handler: this.showWindow,
+                scope: this
+            });
+        }
+    },
+
+
+    /*
+     * Method: _createCbSearch
+     * Returns: {Ext.form.ComboBox}
+     * Create combo use to search free text input by user and display proposition (autocompletion)
+     */
+    _createCbSearch: function() {
+
+        // get options from config.json file
+        var banGeocoderOptions = this.options;
+
+        // create store to pass free text and get result from service (URL)
+        var store = new Ext.data.JsonStore({
+            proxy: new Ext.data.HttpProxy({
+                url: banGeocoderOptions.geocodeServiceUrl, // set service URL in manifest.json file, more informations in README
```

Comment:
user may modify service url in config.json, not manifest.json.
manifest is written by addon maintainer.
config is under the responsability of the platform administrator